### PR TITLE
improve the documentation on accessors

### LINKED ISSUE #2788: Documentation on accessors?
I would like to extend the XArray library, but before I start, I am looking for more information.  With Panda's Accessor pattern and DataArray, there was a defined pattern to extend the API.

What methods and properties must be implemented in DataArray to assure it works properly?  Is there a full template/example of doing this?

```http://xarray.pydata.org/en/stable/internals.html#extending-xarray``` documentation seems to say you can create a DataSet, but doesn't show what is required for a dataset.  Is there a module I can look at in the repository to show how this is done?

### PR DESCRIPTION
**Pull request by keewis**
We currently have a example for a accessor class in our `internals` page, but we didn't explicitly say what we require of these classes. This PR adds that information, and also fixes the displayed text for the link to `sphinx-autosummary-accessors`.

 - [x] Closes #2788
 - [x] User visible changes (including notable bug fixes) are documented in `whats-new.rst`

### CODE REVIEW

**Comment by mathause** [comment]
LGTM, I tried to come up with a sentence without the double "only" but wasn't happy either :wink:

File: doc/internals.rst [hunk 1]
```diff
@@ -116,6 +116,10 @@ xarray:
 
 .. literalinclude:: examples/_code/accessor_example.py
 
+In general, the only requirement is that the accessor's ``__init__`` only has a single
```
Comment (dcherian) [inline doc/internals.rst:119]:
```suggestion
In general, the accessor's ``__init__`` must only have a single
```

File: doc/internals.rst [hunk 2]
```diff
@@ -116,6 +116,10 @@ xarray:
...
+parameter except from ``self``: the ``Dataset`` or ``DataArray`` object it is supposed
+to work on.
```
Comment (keewis) [inline doc/internals.rst:121]: thanks for the hint, @mathause, I didn't notice that when writing it. How about this:
```suggestion
In general, the only restriction on the accessor class is that the ``__init__`` method
has just a single parameter: the ``Dataset`` or ``DataArray`` object it is supposed
to work on.
```

**Review by dcherian — APPROVED** [review]

File: doc/internals.rst (hunk 2 above)
Comment (keewis) [reply doc/internals.rst:121]: Including @dcherian's suggestion, maybe something like this:
```suggestion
In general, the only restriction on the accessor class is that the ``__init__`` method
must have a single parameter: the ``Dataset`` or ``DataArray`` object it is supposed
to work on.
```

Comment (dcherian) [reply doc/internals.rst:121]: The second one looks good.

**Comment by keewis** [comment]
thanks for the reviews, @dcherian, @mathause.

**Pull request merged**